src/event_form/steps/event-form-step3.controller.js

```diff
@@ -226,87 +225,68 @@ function EventFormStep3Controller(
+    // exit code when the filterValue has < 3 characters
+    if (filterValue.length < 3) {
+      $scope.locationsSearched = false;
+      return;
     }
 
+    $scope.locationsSearched = true;
+
     function updateLocationsAndReturnList (locations) {
       // Loop over all locations to check if location is translated.
       _.each(locations, function(location, key) {
         locations[key] = jsonLDLangFilter(locations[key], language, true);
       });
-      $scope.locationsForCity = locations;
-      return locations;
+      // sort locations by Levenshtein distance
+      var sortedLocations = locations.sort(orderSearchedLocations(filterValue));
+      $scope.locationsForCity = sortedLocations;
+      return sortedLocations;
     }
 
-    function clearLoadingState() {
-      $scope.locationsSearched = false;
-      $scope.loadingPlaces = false;
+    function showErrorAndReturnEmptyList () {
+      $scope.locationAutoCompleteError = true;
+      return [];
     }
 
-    $scope.loadingPlaces = true;
-    $scope.locationAutoCompleteError = false;
-
     if ($scope.selectedCountry.code === 'BE') {
       return cityAutocomplete
-        .getPlacesByZipcode(city.zip, 'BE')
-        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList)
-        .finally(clearLoadingState);
+        .getPlacesByZipcode($scope.selectedCityObj.zip, 'BE', filterValue)
+        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList);
     }
+
     if ($scope.selectedCountry.code === 'NL') {
       return cityAutocomplete
-        .getPlacesByCity(city.label, 'NL')
-        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList)
-        .finally(clearLoadingState);
+        .getPlacesByCity($scope.selectedCityObj.name, 'NL')
+        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList);
     }
+
   };
 
+  $scope.getPlaces = controller.getPlaces;
+
   controller.cityHasLocations = function () {
     return $scope.locationsForCity instanceof Array && $scope.locationsForCity.length > 0;
   };
-  $scope.cityHasLocations = controller.cityHasLocations;
-
-  controller.locationSearched = function () {
-    $scope.locationsSearched = true;
-  };
-  $scope.locationSearched = controller.locationSearched;
 
-  controller.filterCityLocations = function (filterValue) {
-    return function (location) {
-      var words = filterValue.match(/\w+/g).filter(function (word) {
-        return word.length > 2;
-      });
-      var addressMatches = words.filter(function (word) {
-        return location.address.streetAddress.toLowerCase().indexOf(word.toLowerCase()) !== -1;
-      });
-      var nameMatches = words.filter(function (word) {
-        return location.name.toLowerCase().indexOf(word.toLowerCase()) !== -1;
-      });
-
-      return addressMatches.length + nameMatches.length >= words.length;
-    };
-  };
-  $scope.filterCityLocations = controller.filterCityLocations;
+  /**
+   * Order locations by Levenshtein distance
```

Comment:
Maybe we can remove this comment and the one on line [248](https://github.com/cultuurnet/udb3-angular/pull/541/files#diff-7607195259faf192fac0134b5a31f415R248) if we can include the sorting algorithm name in the function name? Something like `orderSearchedLocationsByLevenshteinDistance()`? (Just a suggestion as an example - feel free to pick another name if you think there's a better one.)